$NetBSD: patch-configure,v 1.39 2019/10/04 10:33:21 maya Exp $

All kinds of OS-specific changes to configure
Honor LDFLAGS.

Add SunOS -lnsl -lsocket
Recognise SunOS/amd64 as amd64
Recognise SunOS as having shared libraries

--- configure.orig	2019-08-05 17:32:44.000000000 +0000
+++ configure
@@ -13311,7 +13311,7 @@ sharedlib_cflags=''
 mksharedlib='shared-libs-not-available'
 rpath=''
 mksharedlibrpath=''
-natdynlinkopts=""
+natdynlinkopts="-L/usr/lib -R/usr/lib -R/usr/pkg/lib"
 
 if test x"$enable_shared" != "xno"; then :
   case $host in #(
@@ -13347,9 +13347,14 @@ esac ;; #(
       rpath="-Wl,-rpath,"
       mksharedlibrpath="-Wl,-rpath,"
       natdynlinkopts="-Wl,-E"
-      shared_libraries_supported=true ;; #(
+      shared_libraries_supported=true
+    *-*-solaris* ;; #(
   *) :
-     ;;
+    sharedlib_cflags="-fPIC"
+      mksharedlib="$CC -shared"
+      rpath="-Wl,-rpath,"
+      mksharedlibrpath="-Wl,-rpath,"
+      shared_libraries_supported=true ;;
 esac
 fi
 
@@ -13403,6 +13408,8 @@ fi ;; #(
     natdynlink=true ;; #(
   x86_64-*-netbsd*) :
     natdynlink=true ;; #(
+  x86_64-*-solaris*) :
+    natdynlink=true ;; #(
   i386-*-gnu0.3) :
     natdynlink=true ;; #(
   i[3456]86-*-haiku*) :
@@ -13415,6 +13422,8 @@ fi ;; #(
     natdynlink=true ;; #(
   aarch64-*-linux*) :
     natdynlink=true ;; #(
+  aarch64-*-netbsd*) :
+    natdynlink=true ;; #(
   *) :
      ;;
 esac
@@ -13466,6 +13475,8 @@ case $host in #(
     arch=i386; system=beos ;; #(
   i[3456]86-*-cygwin) :
     arch=i386; system=cygwin ;; #(
+	i[3456]86-*-interix3) :
+		arch=i386; system=interix3 ;; #(
   i[3456]86-*-darwin*) :
     if $arch64; then :
   arch=amd64
@@ -13510,6 +13521,10 @@ fi; system=elf ;; #(
     arch=arm; model=armv6; system=linux_eabi ;; #(
   armv6*-*-freebsd*) :
     arch=arm; model=armv6; system=freebsd ;; #(
+  armv6*-*-netbsd*) :
+    arch=arm; model=armv6; system=netbsd ;; #(
+  armv7*-*-netbsd*) :
+    arch=arm; model=armv7; system=netbsd ;; #(
   earmv6*-*-netbsd*) :
     arch=arm; model=armv6; system=netbsd ;; #(
   earmv7*-*-netbsd*) :
@@ -13544,8 +13559,12 @@ fi; system=elf ;; #(
     arch=amd64; system=mingw64 ;; #(
   aarch64-*-linux*) :
     arch=arm64; system=linux ;; #(
+  aarch64-*-netbsd*) :
+    arch=arm64; system=netbsd ;; #(
   x86_64-*-cygwin*) :
-    arch=amd64; system=cygwin
+    arch=amd64; system=cygwin ;; #(
+  x86_64-*-solaris*) :
+    arch=amd64; system=solaris
  ;; #(
   *) :
      ;;
@@ -13884,7 +13903,7 @@ fi
 # for the moment, to be backward-compatible
 
 case $host in #(
-  *-*-mingw32|*-pc-windows) :
+  *-*-mingw32|*-pc-windows|*-*-netbsd*) :
      ;; #(
   *) :
     has_c99_float_ops=true
@@ -14085,6 +14104,9 @@ case $host in #(
   *-*-mingw32|*-pc-windows) :
     cclibs="$cclibs -lws2_32"
     sockets=true ;; #(
+  *-*-solaris*) :
+    cclibs="$cclibs -lnsl -lsocket"
+    sockets=true ;; #(
   *) :
 
     ac_fn_c_check_func "$LINENO" "socket" "ac_cv_func_socket"
